src/Chapter1.hs

```diff
@@ -653,8 +669,11 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
-
+firstDigit :: Int -> Int
+firstDigit n
+    | m < 10 = m
+    | otherwise = firstDigit (div (m - mod m 10) 10)
+    where m = abs n -- this really only needs to be applied once at the start
```

Comment:
This is an absolutely fair point! 🎯 
 Currently, it is not the case, but it could be fixed without too much changes. You can move your current implementation to the helper function in `where` construction. And the top-level `firstDigit` function should call it with the absolute value. Something like this:
 
 
```haskell
firstDigit n = go (abs n)
  where
    go :: Int -> Int
    go m =
        | m < 10 = m
        | otherwise = go (div m 10)
```

Let me know if this makes sense 🙂

Reply:
Yep that makes sense, thanks!